Assets/Scripts/SS3D/Systems/Inventory/Items/ItemActor.cs

```diff
@@ -226,46 +235,44 @@ public override void CreateSourceInteractions(IInteractionTarget[] targets, List
             interactions.Add(new InteractionEntry(null, dropInteraction));
         }
 
-        public bool InContainer()
+        /// <summary>
+        /// Checks if the item is currently stored in a container
+        /// </summary>
+        /// <returns></returns>
+        public bool InOnContainer()
         {
-            return _container != null;
+            return item.IsOnContainer();
         }
 
+        /// <summary>
+        /// Checks if the item has an specific trait
+        /// </summary>
+        /// <param name="trait"></param>
+        /// <returns></returns>
         public bool HasTrait(Trait trait)
         {
-            return traits.Contains(trait);
-        }
-
-        public bool HasTrait(string name)
-        {
-            var hash = Animator.StringToHash(name.ToUpper());
-            foreach (Trait trait in traits)
-            {
-                if (trait.Hash == hash)
-                    return true;
-            }
-            return false;
+            return item.HasTrait(trait);
         }
 
         [Server]
         public void SetContainer(Container newContainer, bool alreadyAdded, bool alreadyRemoved)
         {
-            if (_container == newContainer)
+            if (Container == newContainer)
```

Comment:
that logic can probably go in item

Reply:
The logic is server-side, meaning the Item class would be calling Server functions and ServerRPCs, kinda breaks the Humble Object Pattern in my opinion

Reply:
What I mean is something like this : 
in itemActor
```
         [Server]
        public void SetContainer(Container newContainer, bool alreadyAdded, bool alreadyRemoved)
        {
            bool containerSet = item.SetContainer(newContainer, alreadyAdded, alreadyRemoved);
            if (containerSet) RpcSetContainer(newContainer);
        }
```

in item
```
        public bool SetContainer(Container newContainer, bool alreadyAdded, bool alreadyRemoved)
        {
            if (Container == newContainer)
            {
                return false;
            }

            if (Container != null)
            {
                Container.RemoveItem(this);
            }

            if (!alreadyAdded && newContainer != null)
            {
                newContainer.AddItem(this);
            }

            item.Container = newContainer;
            return true;
        }
```